### PR TITLE
Add support for Somfy RTS

### PR DESCRIPTION
**Pull request by matthiasseemoo**
Added decoder for Somfy RTS. Tests are submitted in https://github.com/merbanan/rtl_433_tests/pull/363.

### CODE REVIEW

File: src/devices/somfy.c [hunk 1]
```diff
@@ -0,0 +1,178 @@
+/** @file
+    Somfy RTS
```
Comment (zuckschwerdt) [inline src/devices/somfy.c:2]: End first sentence with a dot "."

File: src/devices/somfy.c [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+*/
+
+#include "decoder.h"
+
+/*
+    Protocol description:
```
Comment (zuckschwerdt) [inline src/devices/somfy.c:15]: Repeat the first sentence here.
Do not indent.
Use valid markdown, e.g. verbatim blocks indented by 4 spaces.

File: src/devices/somfy.c [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    "Sun + Flag (9)",
+    "Flag (10)",
+    "? (11)",
+    "? (12)",
+    "? (13)",
+    "? (14)",
+    "? (15)",
+};
+
+static int somfy_rts_callback(r_device *decoder, bitbuffer_t *bitbuffer) {
```
Comment (zuckschwerdt) [inline src/devices/somfy.c:71]: function opening brace on next line.

File: src/devices/somfy.c [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    // extract control
+    control = (message_bytes[1] & 0xf0) >> 4;
+
+    // extract counter independent of system's byte order
+    counter = message_bytes[3] + message_bytes[2] * 255;
+
+    // extract address bytes into hexstring
+    snprintf(address, sizeof(address), "%02x%02x%02x", message_bytes[4], message_bytes[5], message_bytes[6]);
+
+    data = data_make(
```
Comment (zuckschwerdt) [inline src/devices/somfy.c:139]: enclose with `/* clang-format off */` `/* clang-format on */`

File: src/devices/somfy.c [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+
+static char *output_fields[] = {
+    "model",
+    "random",
+    "control",
+    "checksum",
+    "counter",
+    "address",
+    "retransmission",
+    NULL
```
Comment (zuckschwerdt) [inline src/devices/somfy.c:161]: add trailing comma "," and indent twice (8 spaces)

File: src/devices/somfy.c [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+};
+
+// rtl_433 -r g001_433.414M_250k.cu8 -X "n=somfy-test,m=OOK_PCM,s=604,l=604,t=40,r=10000,g=3000,y=2416"
+
+r_device somfy_rts = {
+    .name           = "Somfy RTS",
```
Comment (zuckschwerdt) [inline src/devices/somfy.c:167]: indent twice (8 spaces)
Perhaps "Somy RTS remote control"?

File: src/devices/somfy.c [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+    bitbuffer_extract_bytes(&bitbuffer_decoded, 0, 0, message_bytes, 56);
+
+    // descramble
+    for (int i = 6; i >= 0; i--)
+        message_bytes[i] = message_bytes[i] ^ ((i == 0) ? 0 : message_bytes[i-1]);
+
+    // calculate checksum
+    for (int i = 0; i < 7; i++)
+        chksum ^= (message_bytes[i] & 0xf) ^ (message_bytes[i] >> 4);
+
```
Comment (zuckschwerdt) [inline src/devices/somfy.c:126]: `return DECODE_FAIL_MIC` on fail

File: src/devices/somfy.c [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+    counter = message_bytes[3] + message_bytes[2] * 255;
+
+    // extract address bytes into hexstring
+    snprintf(address, sizeof(address), "%02x%02x%02x", message_bytes[4], message_bytes[5], message_bytes[6]);
+
+    data = data_make(
+            "model",          "",               DATA_STRING, _X("Somfy-RTS","Somfy RTS"),
+            "random",         "Random",         DATA_INT,    random,
+            "control",        "Control",        DATA_STRING, control_str[control],
+            "checksum",       "Checksum",       DATA_STRING, chksum == 0 ? "OK" : "NOK",
```
Comment (zuckschwerdt) [inline src/devices/somfy.c:143]: don't output bad packets.

File: src/devices/somfy.c [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+            "counter",        "Counter",        DATA_INT,    counter,
+            "address",        "Address",        DATA_STRING, address,
+            "retransmission", "Retransmission", DATA_STRING, is_retransmission ? "TRUE" : "FALSE",
+            NULL);
```
Comment (zuckschwerdt) [inline src/devices/somfy.c:147]: add `"mic",              "Integrity",    DATA_STRING, "CHECKSUM",`

File: src/devices/somfy.c [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+    control = (message_bytes[1] & 0xf0) >> 4;
+
+    // extract counter independent of system's byte order
+    counter = message_bytes[3] + message_bytes[2] * 255;
+
+    // extract address bytes into hexstring
+    snprintf(address, sizeof(address), "%02x%02x%02x", message_bytes[4], message_bytes[5], message_bytes[6]);
+
+    data = data_make(
+            "model",          "",               DATA_STRING, _X("Somfy-RTS","Somfy RTS"),
```
Comment (zuckschwerdt) [inline src/devices/somfy.c:140]: use `DATA_STRING, "Somfy-RTS",`

File: src/devices/somfy.c (hunk 6 above)
Comment (matthiasseemoo) [reply src/devices/somfy.c:167]: I would stay with "Somfy RTS" as there are other products besides remote controls that also use that protocol (https://asset.somfy.com/Document/4e5d6a89-a67c-4c7e-b303-3cd59a982e9a_RTS_Pocket_Guide_6-2019.pdf). For example, sun and wind sensors. I just do not have them to generate sample traces.

File: src/devices/somfy.c [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+    "retransmission",
+    NULL
+};
+
+// rtl_433 -r g001_433.414M_250k.cu8 -X "n=somfy-test,m=OOK_PCM,s=604,l=604,t=40,r=10000,g=3000,y=2416"
+
+r_device somfy_rts = {
+    .name           = "Somfy RTS",
+    .modulation     = OOK_PULSE_PCM_RZ,
+    .short_width    = 604,   // short pulse is ~604 us
```
Comment (zuckschwerdt) [inline src/devices/somfy.c]: this could be annotated with e.g.
nominal bit width is 604 us
NRZ, short=long

File: src/devices/somfy.c [hunk 12]
```diff
@@ -0,0 +1,193 @@
...
+        return DECODE_FAIL_SANITY;
+
+    if (bitbuffer_manchester_decode(bitbuffer, decode_row, data_start, &bitbuffer_decoded, 56) - data_start < 56)
+        if (bitbuffer_manchester_decode(bitbuffer, decode_row, data_start - 1, &bitbuffer_decoded, 56) - data_start + 1 < 56)
+            return DECODE_FAIL_MIC;
+
+    bitbuffer_extract_bytes(&bitbuffer_decoded, 0, 0, message_bytes, 56);
+
+    // descramble
+    for (int i = 6; i >= 0; i--)
```
Comment (zuckschwerdt) [inline src/devices/somfy.c:124]: with `i > 0` you don't need to check for `i == 0`

File: src/devices/somfy.c [hunk 13]
```diff
@@ -0,0 +1,193 @@
...
+        message_bytes[i] = message_bytes[i] ^ ((i == 0) ? 0 : message_bytes[i-1]);
+
+    // calculate checksum
+    for (int i = 0; i < 7; i++)
```
Comment (zuckschwerdt) [inline src/devices/somfy.c:128]: The intent will be clearer with
chk = xor_bytes(message_bytes, 7);
chk = (chk & 0xf) ^ (chk >> 4); // fold to nibble

File: src/devices/somfy.c [hunk 14]
```diff
@@ -0,0 +1,193 @@
...
+    }
+
+    if (data_start == 0)
+        return DECODE_FAIL_SANITY;
+
+    if (bitbuffer_search(bitbuffer, decode_row, 0, preamble_pattern, preamble_pattern_bit_length) != 0)
+        return DECODE_FAIL_SANITY;
+
+    if (bitbuffer_manchester_decode(bitbuffer, decode_row, data_start, &bitbuffer_decoded, 56) - data_start < 56)
+        if (bitbuffer_manchester_decode(bitbuffer, decode_row, data_start - 1, &bitbuffer_decoded, 56) - data_start + 1 < 56)
```
Comment (zuckschwerdt) [inline src/devices/somfy.c:118]: if you consider the `bitbuffer_search` value you don't need to try `-1` here?

Comment (matthiasseemoo) [reply src/devices/somfy.c:118]: I could also write: `if (bitbuffer_manchester_decode(bitbuffer, decode_row, data_start - 1, &bitbuffer_decoded, 56) - (data_start - 1) < 56)`.
Is it what you mean?

File: src/devices/somfy.c [hunk 15]
```diff
@@ -0,0 +1,193 @@
...
+
+    uint8_t message_bytes[7] = { 0 };
+    uint8_t chksum_calc = 0;
+    uint8_t chksum = 0;
+    uint16_t counter = 0;
+    char address[7];
+    uint8_t control = 0;
+    uint8_t seed = 0;
+
+    for (int i = 0; i < bitbuffer->num_rows; i++) {
```
Comment (zuckschwerdt) [inline src/devices/somfy.c:93]: isn't the logic here:
any row > 170 -> retransmission, pick a row, always align at 0xf0f0ff (preamble)?

File: src/devices/somfy.c (hunk 14 above)
Comment (zuckschwerdt) [reply src/devices/somfy.c:118]: I meant `data_start = bitbuffer_search()` ;)

Comment (matthiasseemoo) [reply src/devices/somfy.c:118]: Sorry, it seems like the wrong code portion was displays here. I expect `bitbuffer_search` to return 0 if the preamble is found at the beginning of the bitbuffer. How can `bitbuffer_search` return `-1`?

File: src/devices/somfy.c [hunk 16]
```diff
@@ -0,0 +1,194 @@
...
+
+    // extract counter independent of system's byte order
+    counter = message_bytes[3] + message_bytes[2] * 255;
+
+    // extract address bytes into hexstring
+    snprintf(address, sizeof(address), "%02X%02X%02X", message_bytes[4], message_bytes[5], message_bytes[6]);
+
+    /* clang-format off */
+    data = data_make(
+            "model",          "",               DATA_STRING, "Somfy-RTS",
```
Comment (zuckschwerdt) [inline src/devices/somfy.c:152]: we always try hard to define an id (it's used by default in e.g. mqtt), is address possibly the id here? or is that a variable "target address"?

Comment (matthiasseemoo) [reply src/devices/somfy.c:152]: Address is a fixed address of the transmitter. We could use this as an id. Should I just replace address by id?

File: src/devices/somfy.c (hunk 14 above)
Comment (zuckschwerdt) [reply src/devices/somfy.c:118]: Sorry, I see. The preamble is 24 bits, but you also try to decode with one bit ignored (.i.e +1) after the preamble. Are there both types of transmissions ("last low bit ok/missing" in your rtl_433_tests PR? I'd like to take a peek :)

File: src/devices/somfy.c (hunk 16 above)
Comment (matthiasseemoo) [reply src/devices/somfy.c:152]: Btw. do you have some documentation where you describe all of those requirements? Such as always defining an id, ...

Comment (zuckschwerdt) [reply src/devices/somfy.c:152]: That would be great, but it needs to be plain `int`.
We are a bit light on documentation, this explains some expectations: https://triq.org/rtl_433/DATA_FORMAT.html

File: src/devices/somfy.c (hunk 14 above)
Comment (matthiasseemoo) [reply src/devices/somfy.c:118]: I added it now.

File: src/devices/somfy.c (hunk 16 above)
Comment (matthiasseemoo) [reply src/devices/somfy.c:152]: I can convert the address into an integer, no problem.

Comment (matthiasseemoo) [reply src/devices/somfy.c:152]: but according to the documentation it should be a string.

Comment (zuckschwerdt) [reply src/devices/somfy.c:152]: We generally don't offer a specific "presentation" of data, just raw values. There is sometimes a valid need to do it different, but with 200 decoders we need to consider carefully to preserve some sanity.

Comment (matthiasseemoo) [reply src/devices/somfy.c:152]: How large can the id be? the address is a 24 bit value. However, after using it as Id, in the output, I only see an 8 bit number.

Comment (zuckschwerdt) [reply src/devices/somfy.c:152]: 31 bit should always be safe, so `(message_bytes[4] << 16) | (message_bytes[5] << 8) | message_bytes[6]` should work.

Comment (matthiasseemoo) [reply src/devices/somfy.c:152]: you are right, I should not add 8 bit number and expect a 32 bit output ;-). Now it works.

File: src/devices/somfy.c (hunk 15 above)
Comment (matthiasseemoo) [reply src/devices/somfy.c:94]: We could always align on this preamble, but I also wanted to explicitly check the longer preamble of the retransmissions.

**Comment by matthiasseemoo** [comment]
Can we merge it like this?

**Comment by zuckschwerdt** [comment]
There is quite some jitter in the signal, the bit might not be missing but just wrongly detected. I.e. if you drop the OOK on https://triq.org/pdv/ both g015 and g006 look roughly the same.
(`rtl_433 -w g015_433.414M_250k.ook g015_433.414M_250k.cu8` `rtl_433 -w g006_433.414M_250k.ook 
g006_433.414M_250k.cu8`)

jitter

**Comment by zuckschwerdt** [comment]
The code looks great. Let me figure out why that bit is sliced badly, perhaps there is no need for the added logic here then.
Btw. there is no `sync` for PCM, the comment is useful, but the field likely should not be set in the r_device.

**Comment by zuckschwerdt** [comment]
Yeah, including the pulse when slicing the gap for PCM skews the result. https://github.com/merbanan/rtl_433/blob/master/src/pulse_demod.c#L124
I need to consider why this could be useful and likely change it. Then you don't need that hack for the missing bit.

**Comment by zuckschwerdt** [comment]
The "missing bit after very long pulse" in PCM has been addressed with 97fdf16. Can you rebase and test?

File: src/devices/somfy.c (hunk 14 above)
Comment (matthiasseemoo) [reply src/devices/somfy.c]: With the fix from https://github.com/merbanan/rtl_433/commit/97fdf16333a5f1a7fb5490996a942186be5cf99e my workaround is not needed anymore. Hence, I removed the second call to the Manchester decoder.

**Comment by matthiasseemoo** [comment]
So, finally I rebased by branch and verified that it now also works without the second call to the Manchester decoder. I also fixed the ID's endianess so that the two ID's in the test captures are just off by 1.

You can merge it now.

File: src/devices/somfy.c (hunk 14 above)
Comment (zuckschwerdt) [reply src/devices/somfy.c]: The preamble should then likely be 0xe1e1fe (one 0-bit added) and bitbuffer_search() will yield the proper data_start. Or perhaps use 0xf0f0ff00 with a preamble_len of 25.

**Comment by zuckschwerdt** [comment]
Thanks for the great work, really looks good now. Look into the preamble note if you have time, but that's not a blocker.

**Comment by matthiasseemoo** [comment]
Btw. I only added my C file into the CMake related files. I saw that there are also VC files for the Windows builds. Should I change them as well or do you do this. I cannot test the VC build on my side, though.

**Comment by matthiasseemoo** [comment]
Besides the capture that I generated with rtl_433 using an rtl-sdr, I also generated a capture of the signal using the universal radio hacker (URH) and a HackRF. There, I set the sample rate to 8 Msps. However, when I try to analyze the file using rtl_433, all pulses are recognized as being too long. Changing the sample rate in the file name does not change the detected pulse lengths so it seems like the sample rate is not taken from the file name? Do I need to configure the sample rate manually in this case?

**Comment by zuckschwerdt** [comment]
To fix all build files run `./maintainer_update.py` if you like, but I can do that on merge also.

The sample rate is currently not read back from the filename, planned and prepared though. `-s8M` is needed.

**Comment by matthiasseemoo** [comment]
Running the tool with incorrect sample rate at least lists some pulses:
```
rtl_433 -R167 -r ../rtl_433_tests/tests/somfy/situo_5_rts_pure_ii/01/g001_433.414M_8000k.cu8 -vvvv
rtl_433 version 20.02-164-ga8aa328 branch master at 202009172208 inputs file rtl_tcp RTL-SDR SoapySDR
Use -h for usage help and see https://triq.org/ for documentation.
Trying conf file at "rtl_433.conf"...
Trying conf file at "/Users/matthias/.config/rtl_433/rtl_433.conf"...
Trying conf file at "/usr/local/etc/rtl_433/rtl_433.conf"...
Trying conf file at "/etc/rtl_433/rtl_433.conf"...
Registered 1 out of 167 device decoding protocols
Test mode active. Reading samples from file: ../rtl_433_tests/tests/somfy/situo_5_rts_pure_ii/01/g001_433.414M_8000k.cu8
Input format: CU8 IQ (2ch uint8)
Pulse data: 1 pulses
[  0] Pulse: 83449, Gap: 25001, Period: 108450
Pulse data: 44 pulses
[  0] Pulse: 19999, Gap: 20250, Period: 40249
[  1] Pulse: 19993, Gap: 20218, Period: 40211
[  2] Pulse: 38645, Gap: 10256, Period: 48901
[  3] Pulse: 10338, Gap: 10282, Period: 20620
[  4] Pulse: 10341, Gap: 5120, Period: 15461
[  5] Pulse: 5195, Gap: 5113, Period: 10308
[  6] Pulse: 5200, Gap: 5127, Period: 10327
[  7] Pulse: 5186, Gap: 5122, Period: 10308
[  8] Pulse: 5194, Gap: 10286, Period: 15480
[  9] Pulse: 5186, Gap: 5130, Period: 10316
[ 10] Pulse: 5185, Gap: 5123, Period: 10308
[ 11] Pulse: 10349, Gap: 10291, Period: 20640
[ 12] Pulse: 5179, Gap: 5124, Period: 10303
[ 13] Pulse: 10346, Gap: 10297, Period: 20643
[ 14] Pulse: 5175, Gap: 5145, Period: 10320
[ 15] Pulse: 5170, Gap: 5131, Period: 10301
[ 16] Pulse: 5184, Gap: 5132, Period: 10316
[ 17] Pulse: 10342, Gap: 10285, Period: 20627
[ 18] Pulse: 5186, Gap: 5128, Period: 10314
[ 19] Pulse: 10344, Gap: 10287, Period: 20631
[ 20] Pulse: 5184, Gap: 5128, Period: 10312
[ 21] Pulse: 10344, Gap: 5111, Period: 15455
[ 22] Pulse: 5202, Gap: 5124, Period: 10326
[ 23] Pulse: 5188, Gap: 5128, Period: 10316
[ 24] Pulse: 5185, Gap: 10283, Period: 15468
[ 25] Pulse: 5187, Gap: 5135, Period: 10322
[ 26] Pulse: 5180, Gap: 5124, Period: 10304
[ 27] Pulse: 10342, Gap: 5110, Period: 15452
[ 28] Pulse: 5201, Gap: 10275, Period: 15476
[ 29] Pulse: 10343, Gap: 10277, Period: 20620
[ 30] Pulse: 5187, Gap: 5129, Period: 10316
[ 31] Pulse: 5182, Gap: 5120, Period: 10302
[ 32] Pulse: 10347, Gap: 10281, Period: 20628
[ 33] Pulse: 5187, Gap: 5126, Period: 10313
[ 34] Pulse: 5185, Gap: 5135, Period: 10320
[ 35] Pulse: 10335, Gap: 10277, Period: 20612
[ 36] Pulse: 10351, Gap: 10281, Period: 20632
[ 37] Pulse: 10341, Gap: 10280, Period: 20621
[ 38] Pulse: 5186, Gap: 5137, Period: 10323
[ 39] Pulse: 10325, Gap: 10291, Period: 20616
[ 40] Pulse: 5171, Gap: 5116, Period: 10287
[ 41] Pulse: 5192, Gap: 5137, Period: 10329
[ 42] Pulse: 10329, Gap: 10287, Period: 20616
[ 43] Pulse: 4994, Gap: 25001, Period: 29995
Test mode file issued 7 packets
```
However, running it with the correct bandwidth does not list any pulses:
```
rtl_433 -R167 -r ../rtl_433_tests/tests/somfy/situo_5_rts_pure_ii/01/g001_433.414M_8000k.cu8 -vvvv -s8M
rtl_433 version 20.02-164-ga8aa328 branch master at 202009172208 inputs file rtl_tcp RTL-SDR SoapySDR
Use -h for usage help and see https://triq.org/ for documentation.
Trying conf file at "rtl_433.conf"...
Trying conf file at "/Users/matthias/.config/rtl_433/rtl_433.conf"...
Trying conf file at "/usr/local/etc/rtl_433/rtl_433.conf"...
Trying conf file at "/etc/rtl_433/rtl_433.conf"...
Registered 1 out of 167 device decoding protocols
Test mode active. Reading samples from file: ../rtl_433_tests/tests/somfy/situo_5_rts_pure_ii/01/g001_433.414M_8000k.cu8
Input format: CU8 IQ (2ch uint8)
Test mode file issued 7 packets
```
I did not commit this test file as it is 1.7 MB.

**Comment by matthiasseemoo** [comment]
Running `./maintainer_update.py` also adds the `pulse_analyer.c` file. Are the entries for this file maybe currently missing on the master?

```
diff --git a/src/Makefile.am b/src/Makefile.am
index e6cb397..ea5b209 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -18,6 +18,7 @@ rtl_433_SOURCES      = abuf.c \
                        optparse.c \
                        output_influx.c \
                        output_mqtt.c \
+                       pulse_analyzer.c \
                        pulse_demod.c \
                        pulse_detect.c \
                        pulse_detect_fsk.c \
diff --git a/vs15/rtl_433.vcxproj b/vs15/rtl_433.vcxproj
index b4d1810..b688946 100644
--- a/vs15/rtl_433.vcxproj
+++ b/vs15/rtl_433.vcxproj
@@ -110,6 +110,7 @@ COPY ..\..\libusb\MS64\dll\libusb*.dll $(TargetDir)</Command>
     <ClInclude Include="..\include\optparse.h" />
     <ClInclude Include="..\include\output_influx.h" />
     <ClInclude Include="..\include\output_mqtt.h" />
+    <ClInclude Include="..\include\pulse_analyzer.h" />
     <ClInclude Include="..\include\pulse_demod.h" />
     <ClInclude Include="..\include\pulse_detect.h" />
     <ClInclude Include="..\include\pulse_detect_fsk.h" />
@@ -142,6 +143,7 @@ COPY ..\..\libusb\MS64\dll\libusb*.dll $(TargetDir)</Command>
     <ClCompile Include="..\src\optparse.c" />
     <ClCompile Include="..\src\output_influx.c" />
     <ClCompile Include="..\src\output_mqtt.c" />
+    <ClCompile Include="..\src\pulse_analyzer.c" />
     <ClCompile Include="..\src\pulse_demod.c" />
     <ClCompile Include="..\src\pulse_detect.c" />
     <ClCompile Include="..\src\pulse_detect_fsk.c" />
@@ -263,6 +265,7 @@ COPY ..\..\libusb\MS64\dll\libusb*.dll $(TargetDir)</Command>
     <ClCompile Include="..\src\devices\simplisafe.c" />
     <ClCompile Include="..\src\devices\smoke_gs558.c" />
     <ClCompile Include="..\src\devices\solight_te44.c" />
+    <ClCompile Include="..\src\devices\somfy.c" />
     <ClCompile Include="..\src\devices\springfield.c" />
     <ClCompile Include="..\src\devices\steelmate.c" />
     <ClCompile Include="..\src\devices\tfa_30_3196.c" />
diff --git a/vs15/rtl_433.vcxproj.filters b/vs15/rtl_433.vcxproj.filters
index 41c5869..dcb792e 100644
--- a/vs15/rtl_433.vcxproj.filters
+++ b/vs15/rtl_433.vcxproj.filters
@@ -160,6 +160,9 @@ <U+FEFF><?xml version="1.0" encoding="utf-8"?>
     <ClCompile Include="..\src\output_mqtt.c">
       <Filter>Source Files</Filter>
     </ClCompile>
+    <ClCompile Include="..\src\pulse_analyzer.c">
+      <Filter>Source Files</Filter>
+    </ClCompile>
     <ClCompile Include="..\src\pulse_demod.c">
       <Filter>Source Files</Filter>
     </ClCompile>
@@ -526,6 +529,9 @@ <U+FEFF><?xml version="1.0" encoding="utf-8"?>
     <ClCompile Include="..\src\devices\solight_te44.c">
       <Filter>Source Files\devices</Filter>
     </ClCompile>
+    <ClCompile Include="..\src\devices\somfy.c">
+      <Filter>Source Files\devices</Filter>
+    </ClCompile>
     <ClCompile Include="..\src\devices\springfield.c">
       <Filter>Source Files\devices</Filter>
     </ClCompile>
```

**Comment by zuckschwerdt** [comment]
Oops, yes, I forgot to add that. Just include it, we'll merge this next anyway :)

**Comment by matthiasseemoo** [comment]
done, happy merging :-)

**Comment by zuckschwerdt** [comment]
Can you push again so the new style checks can run?
Preferable add a commit which removes `checksum` and `seed` from the output as those don't contain any useful data. You can add a if-verbose-fprintf for those if you like.

**Comment by matthiasseemoo** [comment]
It would be nice to get some line numbers for those errors:
 Style Check0s
##[error]STDOUT line
Run ./.github/actions/style-check
##[error]NON-ASCII character error
##[error]TRAILING whitespace error
##[error]STDOUT line

**Comment by matthiasseemoo** [comment]
I found it under actions:
 Check failure on line 25 in src/devices/somfy.c

@github-actions
github-actions
/ Check code style

src/devices/somfy.c#L25
NON-ASCII character error
 Check failure on line 43 in src/devices/somfy.c

@github-actions
github-actions
/ Check code style

src/devices/somfy.c#L43
TRAILING whitespace error

**Comment by matthiasseemoo** [comment]
Done ;-)

**Comment by zuckschwerdt** [comment]
Should also be visible with pretty annotations on your commit: https://github.com/merbanan/rtl_433/pull/1496/files/cf953618d61caa6b2ffe7bb549a71f53cc8196fb..b858ea34a4cd55f100d0bae5118ad95b46c4545c
Thanks being the guinea pig here ;)

**Pull request merged**